src/Features/CSharp/Portable/CodeRefactorings/ConvertToInterpolatedString/ConvertToInterpolatedStringRefactoringProvider.cs

```diff
@@ -0,0 +1,96 @@
+            var stringToken = literalExpression.Token;
+            var text = stringToken.ToString();
+
+            if (stringToken.IsVerbatimStringLiteral())
+            {
+                // We need to escape braces as this is an ambiguous case in interpolated strings that is 
+                // not ambiguous in verbatim strings
+                text = text.Replace(@"'{'", @"'{{'").Replace(@"'}'", @"'}}'");
+            }
+
+            return text;
+        }
+
+        protected override bool IsStringLiteral(LiteralExpressionSyntax firstArgument) =>
+            firstArgument?.IsKind(SyntaxKind.StringLiteralExpression) == true;
+
+        protected override ExpressionSyntax GetExpression(ArgumentSyntax argument) =>
+            argument.Expression;
+
+        protected override SyntaxList<InterpolatedStringContentSyntax> GetInterpolatedStringContents(InterpolatedStringExpressionSyntax interpolatedString) =>
+            interpolatedString.Contents;
+
+        protected override string GetValueText(LiteralExpressionSyntax literalExpression) =>
+            literalExpression.Token.ValueText;
+
+        protected override bool IsNumericLiteralExpression(LiteralExpressionSyntax literalExpression) =>
+            literalExpression?.IsKind(SyntaxKind.NumericLiteralExpression) == true;
+
+        protected override SyntaxNode ReplaceExpressionInInterpolation(InterpolationSyntax node, ExpressionSyntax expressionSyntax) => 
+            node.WithExpression(FixTrivia(expressionSyntax));
+
+        /// <summary>
+        /// Since C# interpolations cannot be on more than one line, we need to remove newlines if possible.
+        /// </summary>
+        public static ExpressionSyntax FixTrivia(ExpressionSyntax node) =>
+            node.ReplaceTokens(node.DescendantTokens(descendIntoTrivia: true), RemoveTriviaForTokens);
+
+        private static SyntaxToken RemoveTriviaForTokens(SyntaxToken originalToken, SyntaxToken rewrittenToken) =>
+            rewrittenToken
+                .WithLeadingTrivia(
+                    ReplaceSyntaxKindsWithElasticSpace(rewrittenToken.LeadingTrivia, SyntaxKind.WhitespaceTrivia, SyntaxKind.EndOfLineTrivia))
+                .WithTrailingTrivia(
+                    ReplaceSyntaxKindsWithElasticSpace(rewrittenToken.TrailingTrivia, SyntaxKind.WhitespaceTrivia, SyntaxKind.EndOfLineTrivia));
+
+        private static SyntaxTriviaList ReplaceSyntaxKindsWithElasticSpace(SyntaxTriviaList trivialList, params SyntaxKind[] kinds) =>
+            trivialList.Select(x => kinds.Any(y => x.IsKind(y)) ? SyntaxFactory.ElasticSpace : x).ToSyntaxTriviaList();
+
+        protected override ExpressionSyntax CastAndParenthesizeExpressionsConvertedType(ArgumentSyntax argument, SemanticModel semanticModel) =>
+            Cast(argument.Expression, semanticModel.GetTypeInfo(argument.Expression).ConvertedType).Parenthesize();
+
+
+        private static ExpressionSyntax Cast(ExpressionSyntax expression, ITypeSymbol targetType) =>
```

Comment:
It looks like this function could be done entirely in the base type.  You just need to use SyntaxGenarator.